internal/core/data/init.go

```diff
@@ -96,5 +89,13 @@ func BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startupTimer star
 		loggingClient.Error(fmt.Sprintf("failed to create messaging client: %s", err.Error()))
 	}
 
+	dic.Update(di.ServiceConstructorMap{
+		dataContainer.MessagingClientName: func(get di.Get) interface{} {
+			return msgClient
+		},
```

Comment:
You're putting a potentially invalid or nil reference for msgClient into the DIC.  Would it be better to fail-fast and stop the service here?

Reply:
Yes, I think failing fast is the best option here.

Reply:
I don't know if it makes sense contextually (or the way the underlying code that returns msgClient is written) -- you might consider implementing multiple retries before failing.  See [here](https://github.com/edgexfoundry/edgex-go/blob/333829e89581c14c68f5ecbcfda36ed5096dfa50/internal/pkg/bootstrap/handlers/database/database.go#L118) for an example that uses the startup timer to determine when to give up.